node/e2e_test.go

```diff
@@ -0,0 +1,89 @@
+package node
+
+import (
+	"context"
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	logruslogger "github.com/virtual-kubelet/virtual-kubelet/log/logrus"
+	"gotest.tools/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var enableEnvTest = flag.Bool("envtest", false, "Enable envtest based tests")
+
+func TestMain(m *testing.M) {
```

Comment:
Maybne these tests should go in their own package?

Reply:
I wanted to do that / thought about doing that, but there's private members of pod controller that we want to prod at while running these tests.

Reply:
`<-node.chReady` seems like the only private member unless I missing something.
We should have a `Ready()` function for this just like pod controller.

Reply:
Okay, I'll try to figure something out.